src/lib.rs

```diff
@@ -207,395 +432,112 @@ impl Mmap {
-        }
-    }
-
-    /// Flushes outstanding view modifications to disk.
-    ///
-    /// When this returns with a non-error result, all outstanding changes to a
-    /// file-backed memory map view are guaranteed to be durably stored. The
-    /// file's metadata (including last modification timestamp) may not be
-    /// updated.
-    pub fn flush(&self) -> Result<()> {
-        self.inner_mut().flush_range(self.offset, self.len)
-    }
-
-    /// Asynchronously flushes outstanding memory map view modifications to
-    /// disk.
-    ///
-    /// This method initiates flushing modified pages to durable storage, but it
-    /// will not wait for the operation to complete before returning. The file's
-    /// metadata (including last modification timestamp) may not be updated.
-    pub fn flush_async(&self) -> Result<()> {
-        self.inner_mut().flush_async_range(self.offset, self.len)
-    }
-
-    /// Returns the length of the memory map view.
-    pub fn len(&self) -> usize {
-        self.len
-    }
-
-    /// Returns a shared pointer to the mapped memory.
-    ///
-    /// See `Mmap::as_slice` for invariants that must hold when dereferencing
-    /// the pointer.
-    pub fn ptr(&self) -> *const u8 {
-        unsafe { self.inner().ptr().offset(self.offset as isize) }
-    }
-
-    /// Returns a mutable pointer to the mapped memory.
-    ///
-    /// See `Mmap::as_mut_slice` for invariants that must hold when
-    /// dereferencing the pointer.
-    pub fn mut_ptr(&mut self) -> *mut u8 {
-        unsafe { self.inner_mut().mut_ptr().offset(self.offset as isize) }
-    }
-
-    /// Returns the memory mapped file as an immutable slice.
-    ///
-    /// ## Unsafety
-    ///
-    /// The caller must ensure that the file is not concurrently modified.
-    pub unsafe fn as_slice(&self) -> &[u8] {
-        &self.inner().as_slice()[self.offset..self.offset + self.len]
-    }
-
-    /// Returns the memory mapped file as a mutable slice.
-    ///
-    /// ## Unsafety
+    /// # Errors
     ///
-    /// The caller must ensure that the file is not concurrently accessed.
-    pub unsafe fn as_mut_slice(&mut self) -> &mut [u8] {
-        &mut self.inner_mut().as_mut_slice()[self.offset..self.offset + self.len]
-    }
-
-    /// Clones the view of the memory map.
+    /// This method returns `Err` when the underlying system call fails, which can happen for
+    /// a variety of reasons, such as when you don't have the necessary permissions for the file.
     ///
-    /// The underlying memory map is shared, and thus the caller must ensure that the memory
-    /// underlying the view is not illegally aliased.
-    pub unsafe fn clone(&self) -> MmapView {
-        MmapView {
-            inner: self.inner.clone(),
-            offset: self.offset,
-            len: self.len,
-        }
+    /// This method will **not** return `Err` if the passed `protection` is mutable.
+    pub fn make_read_only(mut self, protection: Protection) -> Result<Mmap> {
```

Comment:
Perhaps this could be `make_read_only`, and `make_read_execute`?  Are there any other valid outcomes from this?